presto-main/src/test/java/io/prestosql/server/TestQueryResource.java

```diff
@@ -111,6 +149,60 @@ public void testGetQueryInfoExecutionFailure()
         assertEquals(info.getFailureInfo().getErrorCode(), DIVISION_BY_ZERO.toErrorCode());
     }
 
+    @Test
+    public void testCancel()
+    {
+        String queryId = startQuery("SELECT * FROM tpch.sf100.lineitem");
+
+        server.getAccessControl().deny(privilege("query", KILL_QUERY));
+        try {
+            assertEquals(cancelQueryInfo(queryId), 403);
+        }
+        finally {
+            server.getAccessControl().reset();
+        }
+
+        assertEquals(cancelQueryInfo(queryId), 204);
```

Comment:
call this method twice to see how it behaves when user cancel cancelled (or killed) query. Unless it is tested somewhere else.